### ISSUE TITLE
[proc macros]: can't define trait without `client feature` which brings lots of unecessary deps.

### ISSUE THREAD

**Issue by niklasad1** [paritytech/jsonrpsee#649]
I noticed after the refactoring of the `core crate` and if I want to do:

```rust
use jsonrpsee::{core::RpcResult, proc_macros::rpc};

#[rpc(client, server, namespace = "myapi")]
pub trait MyApi {
	#[method(name = "submitBytes")]
	async fn submit_bytes(&self,  bytes: Vec<u8>) -> RpcResult<()>;
}
```

To get the `client traits` required then one must enable the `client feature` which brings in a bunch of dependencies from `http-client` and `ws-client`. 

Maybe we could enable `"jsonrpsee-core/client"` in the macros feature it's really light weight just a few of our own types.

**Comment by dvdplm** [comment]
Makes sense to me.

**Comment by niklasad1** [comment]
Closed by #656
